design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,394 @@
+- Make Velero can support more complicated namespaced and cluster resources filtering scenarios in backup.
+
+## Non Goals
+- Enrich the resource filtering rules, for example, advanced PV filtering and filtering by resource names.
+
+
+## High-Level Design
+Four new parameters are added into command `velero backup create`: `--include-cluster-scope-resources`, `--exclude-cluster-scope-resources`, `--include-namespaced-resources` and `--exclude-namespaced-resources`. 
+`--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` are used to filter cluster scope resources included or excluded in backup per resource type.
+`--include-namespaced-resources` and `--exclude-namespaced-resources` are used to filter namespace scope resources included or excluded in backup per resource type.
+Restore and other code pieces also use resource filtering will be handled in future releases.
+
+### Parameters Rules
+
+* `--include-namespaced-resources` valid value is comma separated string. Not setting the parameter means including all namespace resources.
+
+* `--include-cluster-scope-resources`, `--exclude-cluster-scope-resources` and `--exclude-namespaced-resources` valid value include `*` and comma separated string.
+
+* `--include-cluster-scope-resources`, `--include-namespaced-resources`, `--exclude-cluster-scope-resources` and `--exclude-namespaced-resources` parameters are mutual exclusive with `--include-cluster-resources`, `--include-resources` and `--exclude-resources` parameters. If both sets of parameters are provisioned, validation failure should be returned.
+
+* `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` can only contain cluster scope resource type names. If namespace scope resource type names are included, validation failure should be returned.
+
+* If there are conflicts between `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` specified resources type lists, validation failure should be returned.
+
+* `--include-namespaced-resources` and `--exclude-namespaced-resources` can only contain namespace scope resource type names. If cluster scope resource type names are included, validation failure should be returned.
+
+* If there are conflicts between `--include-namespaced-resources` and `--exclude-namespaced-resources` specified resources type lists, validation failure should be returned.
+
+* If  `--include-namespaced-resources` is not present, it means all namespace scope resources are included per resource type.
+
+* If `--include-cluster-scope-resources` is not present, it means no additional cluster resource is included per resource type. Cluster resources are related to the namespace scope resources, which means those are returned in the namespace resources' BackupItemAction's result AdditionalItems array, are still included in backup by default. If the parameter is set, listed resources and namespaced resources related cluster resources are included in backup.
```

Comment:
Could you elaborate some examples, like `pv` and `crd`

Reply:
Sure. Adding backing PVC will include related PV as the example.

Reply:
"* If  `--include-namespaced-resources` is not present, it means all namespace scope resources are included per resource type."
I don't think this is correct. We should require the include `*`, that way a user can choose to not include any namespaced resources by omitting it.

Reply:
Please see the previous comment and the [overall comment](https://github.com/vmware-tanzu/velero/pull/5333#issuecomment-1410218645).
There is some change in the default value meaning. Please check whether this works for you.